src/cephadm/cephadm

```diff
@@ -5163,6 +5277,123 @@ def list_daemons(ctx, detail=True, legacy_dir=None):
     return ls
 
 
+class ServiceInfo:
+    status: str = ''
+    load: str = ''
+    active: str = ''
+    sub: str = ''
+    image_id: str = ''
+    container_name: str = ''
+    container_id: str = ''
+    container_image: str = ''
+    container_image_id: str = ''
+    container_created: str = ''
+    version: str = ''
+
+    def update(self, **kwargs: Any) -> None:
+        """update the instance with arbitrary k=v pairs
+        """
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+    def get_container_stats(self) -> str:
+        """provide a view of the container stats compatible with get_container_stats method
+
+        Returns:
+            str: container information in a comma separated string
+        """
+        return f'{self.container_id},{self.container_image},{self.container_image_id},{self.container_created},{self.version}'
+
+    def __str__(self) -> str:
+        out = []
+        for k, v in self.__dict__.items():
+            out.append(f'{k}: {v}')
+        return '\n'.join(out)
+
+
+def _build_service_map(ctx: CephadmContext, fsid: str) -> Dict[str, ServiceInfo]:
+    """Create a service map indexed by a systemd unit name
+
+    Args:
+        ctx (CephadmContext): run time context
+        fsid (str): Cluster FSID
+
+    Returns:
+        Dict[str, ServiceInfo]: Dict providing a lookup of systemd unit -> ServiceInfo instance
+    """
+    svc_map = {}
+    st = time.time()
+    not_systemd_dirs = ['home', 'crash', 'selinux']
```

Comment:
this doesn't seem like the right place for this list.. doesn't it need to match the legacy/fallback behavior?  should it be defined elsewhere?